cmds/core/wget/wget_test.go

```diff
@@ -40,131 +43,144 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			if err != nil {
-				// End of test.
 				return
 			}
 			conn.Close()
 		}
 	}()
 
-	h := handler{}
-	go func() {
-		log.Print(http.Serve(l, h))
-	}()
+	return l.Addr().(*net.TCPAddr).Port, l.Close
+}
+
+// TestWget implements a table-driven test.
+func TestWget(t *testing.T) {
+	srv := httptest.NewServer(handler{})
+	defer srv.Close()
+	port, shutdown := setTCPListener(t)
+	defer shutdown()
+
+	var tests = []struct {
+		name        string
+		url         string // in
+		wantContent string // out
+		outputPath  string
+		wantErr     error
+	}{
+		{
+			name:        "ipv4",
+			url:         fmt.Sprintf("%s/200", srv.URL),
+			wantContent: content,
+			outputPath:  "basic",
+			wantErr:     nil,
+		},
+		{
+			name:        "localhost",
+			url:         strings.Replace(srv.URL, "127.0.0.1", "localhost", 1) + "/200",
+			wantContent: content,
+			outputPath:  "ipv4",
+			wantErr:     nil,
+		},
+		// TODO: CircleCI does not support ipv6
+		// {
+		// 	name:    "ipv6",
+		// 	flags:   []string{},
+		// 	url:     "http://[::1]:%[1]d/200",
+		// 	content: content,
+		// 	retCode: 0,
+		// },
+		{
+			name:       "redirect",
+			url:        fmt.Sprintf("%s/302", srv.URL),
+			outputPath: "redirect",
+			wantErr:    nil,
+		},
+		{
+			name:    "4xx error",
+			url:     fmt.Sprintf("%s/404", srv.URL),
+			wantErr: curl.ErrStatusNotOk,
+		},
+		{
+			name:    "5xx error",
+			url:     fmt.Sprintf("%s/500", srv.URL),
+			wantErr: curl.ErrStatusNotOk,
+		},
+		{
+			name:       "connection closed",
+			url:        fmt.Sprintf("http://localhost:%d/200", port),
+			outputPath: "connection-closed",
+			wantErr:    io.EOF,
+		},
+		{
+			name:    "empty url",
+			url:     "",
+			wantErr: errEmptyURL,
+		},
+	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			// Change the working directory to a temporary directory, so we can
-			// delete the temporary files after the test runs.
-			tmpDir := t.TempDir()
-
-			fileName := filepath.Base(tt.url)
-
-			args := append(tt.flags,
-				"-O", filepath.Join(tmpDir, fileName),
-				fmt.Sprintf(tt.url, port, unusedPort))
-			cmd := testutil.Command(t, args...)
-			output, err := cmd.CombinedOutput()
+			tt.outputPath = filepath.Join(t.TempDir(), tt.outputPath)
+			err := newCommand(tt.outputPath, tt.url).run()
 
-			// Check return code.
-			if err := testutil.IsExitCode(err, tt.retCode); err != nil {
-				t.Errorf("exit code: %v, output: %s", err, string(output))
+			if tt.wantErr == nil && err != nil {
+				t.Errorf("expected nil, got: %v", err)
+			}
+			if tt.wantErr != nil && !errors.Is(err, tt.wantErr) {
+				t.Errorf("expected: %v, got: %v", tt.wantErr, err)
 			}
 
-			if tt.content != "" {
-				content, err := os.ReadFile(filepath.Join(tmpDir, fileName))
+			if tt.wantContent != "" {
+				content, err := os.ReadFile(tt.outputPath)
 				if err != nil {
-					t.Errorf("File %s was not created: %v", fileName, err)
+					t.Errorf("file %s was not created: %v", tt.outputPath, err)
```

Comment:
t.Fatalf here?  If `err` is not `nil` then the check of "content" on line 153 won't make any sense.